Dockerfile

```diff
@@ -1,19 +1,24 @@
 FROM ruby:2.4-alpine
 LABEL maintainer="Code Climate <hello@codeclimate.com>"
 
-RUN apk --update add grep
-
 WORKDIR /usr/src/app
 
-COPY Gemfile* ./
-RUN gem install -g
-
 RUN adduser -u 9000 -D app
-RUN chown -R app:app /usr/src/app
-USER app
+
+RUN apk --no-cache add grep
+
+COPY Gemfile* ./
+RUN bundler install --no-cache --without=test --system && \
+  chown -R app:app /usr/local/bundle && \
+  rm -r ~/.bundle
 
 COPY . ./
 
+RUN chown -R app:app . && \
+  ln engine.json /
+
```

Comment:
💄 extra blank line here, we usually only use single blank lines between blocks in dockerfiles.